pokedex/src/Components/App/App.jsx

```diff
@@ -0,0 +1,160 @@
+import React from "react";
+import Header from "../Header/Header";
+import Pokesprites from "../Pokesprites/Pokesprites";
+import Pokeinfo from "../Pokeinfo/Pokeinfo";
+import Aboutme from "../Aboutme/Aboutme";
+import { useState, useEffect } from "react";
+import "../App/App.css";
+import "../style.css";
+import Footer from "../Footer/Footer";
+import { Routes, Route } from "react-router-dom";
+import pokemonData from "../pokemonapi.json";
+import { useNavigate } from "react-router-dom";
+
+const App = () => {
+  const [loading, setLoading] = useState(true);
+  const [url, setUrl] = useState("https://pokeapi.co/api/v2/pokemon/");
+  const [activePokemon, setActivePokemon] = useState();
+  const [pokeInfo, setPokeInfo] = useState({});
+  const [searchTerm, setSearchTerm] = useState("");
+
+  console.log(searchTerm);
+
+  const [pokemonList, setPokemonList] = useState(pokemonData.results);
+  const filteredPokemonList = pokemonList.filter((pokemon) =>
+    pokemon.name.includes(searchTerm)
+  );
+
+  const pokeData = Object.values(pokeInfo).filter((pokemon) => {
+    const bool = pokemon.name.includes(searchTerm);
+    return bool;
+  });
+
+  const navigate = useNavigate();
+
+  const pokeFun = async () => {
+    setLoading(true);
+    const res = await fetch(url);
+    const data = await res.json();
+    fetchPokemonDetails(data.results);
+    setLoading(false);
+  };
+
+  async function getPokemonDetails(url) {
+    try {
+      const res = await fetch(url);
+      if (!res.ok) {
+        throw new Error("Could not get pokemon");
+      }
+      const pokemon = await res.json();
+
+      setActivePokemon(pokemon);
+    } catch (error) {
+      console.error(error);
+    }
+  }
+
+  const fetchPokemonDetails = async (res) => {
+    res.map(async (item) => {
+      const unparsedResult = await fetch(item.url);
+      const result = await unparsedResult.json();
```

Comment:
* You should always check if the response is `ok` according to the theory and previous tasks
* The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special function `checkResponse` that will check any response from the server if it’s ok or not.